### PR TITLE
fluentd: Update to Fluentd v1.16.6/v1.17.1

### PR DESCRIPTION
**Pull request by kenhys**
v1.16.x: stable branch (bundled for fluent-package LTS)

https://github.com/fluent/fluentd-docker-image/commit/ddb5019b433339f9eef991ca373445deb6e4d2b1

v1.17.x: latest branch

https://github.com/fluent/fluentd-docker-image/commit/1055e528ef1acbb073028e0281b60bdaf9a23595

### CODE REVIEW

**Comment by tianon** [comment]
Relevant to https://github.com/docker-library/official-images/pull/15468#issuecomment-2253647480, how do we make sure this gets updated with more regularity?  Is there some way we can assist?  Is it intentional that we're not adding 1.17?

cc @ashie @cosmo0920 (other "recent" PR authors for `library/fluentd` :heart:)

File: library/fluentd [hunk 1]
```diff
@@ -3,13 +3,25 @@ Maintainers: Masahiro Nakagawa <repeatedly@gmail.com> (@repeatedly),
 GitRepo: https://github.com/fluent/fluentd-docker-image.git
 
 # Alpine images
-Tags: v1.16.2-1.1, v1.16-1, latest
+Tags: v1.16.6-1.0, v1.16-1
 Architectures: amd64, arm32v6, arm64v8, i386, ppc64le, s390x
-GitCommit: a4dd65768ec1819574e570716955276c9089326a
+GitCommit: ddb5019b433339f9eef991ca373445deb6e4d2b1
```
Comment (yosifkit) [inline library/fluentd:8]: From the failing test:
```console
 - fluentd:
   - commit ddb5019b433339f9eef991ca373445deb6e4d2b1:
     - is not in the specified ref GitFetch: refs/heads/master
```

It looks like this [commit](https://github.com/fluent/fluentd-docker-image/commit/ddb5019b433339f9eef991ca373445deb6e4d2b1) is in the `v1.16` branch, so these two 16.x blocks would need a `GitFetch: refs/heads/v1.16` to reflect that.

Comment (kenhys) [reply library/fluentd:9]: Thanks! I've overlooked it.

**Review by tianon — APPROVED** [review]
This looks good and is definitely an improvement so I'm going to go ahead and merge, but I'd still like to know more about how we can make sure this keeps getting updated actively in the future. :heart:

**Pull request merged**